src/libraries/System.Security.Cryptography.Cng/src/System/Security/Cryptography/AesCng.cs

```diff
@@ -105,9 +105,21 @@ bool ICngSymmetricAlgorithm.IsWeakKey(byte[] key)
             return false;
         }
 
+        int ICngSymmetricAlgorithm.GetPaddingSize()
+        {
+            return this.GetPaddingSize();
+        }
+
         SafeAlgorithmHandle ICngSymmetricAlgorithm.GetEphemeralModeHandle()
         {
-            return AesBCryptModes.GetSharedHandle(Mode);
+            try
+            {
+                return AesBCryptModes.GetSharedHandle(Mode, FeedbackSize / 8);
+            }
+            catch (NotSupportedException)
+            {
+                throw new CryptographicException(SR.Cryptography_InvalidCipherMode);
```

Comment:
Instead of a normalizing throw here, would it make more sense for AesBCryptModes.GetSharedHandle to throw this exception?

Reply:
To me, this solution looks cleaner. The pattern is that internal calls (GetSharedHandle, CCCrypto, OpenSSL, ..) throws `NotSupportedException` in case of troubles, since their public API already handle invalid scenarios with `CryptographicException`.

Here it is easier to just rewrap the exception than re-implement the check.

YMMV, I'm open to discussion